### ISSUE TITLE
mochaTest  timeout parameter

### ISSUE THREAD

**Issue by vaucouleur** [meanjs/mean#964]
@mleanos  @lirantal 

Hi,

Worth considering: the timeout parameter can be set directly on the options of the mochaTest task. It avoids duplicating the this.timeout(10000) statements everywhere in the mocha tests.

Something like:

    mochaTest: {
      src: testAssets.tests.server,
      options: {
        reporter: 'dot',
        timeout: 10000
      }

See the Option paragraph  of https://github.com/pghalliday/grunt-mocha-test

**Comment by mleanos** [comment]
My original idea was that it would be better to have the *timeouts* set per suite/test. That way when things stabilize with Travis, we can start customizing the timeouts; increasing/decreasing based on the observed performance for each test (or suite). Perhaps, that isn't the best strategy.

In our current situation everything is set to `10000` anyway, so it might make more sense to add this as a global timeout. Eventually, we'll want to decrease this timeout. Once that happens, we could just add a custom timeout to any tests that are giving us *trouble*.

@lirantal WDYT?

Gulp has the same option.. 
https://github.com/sindresorhus/gulp-mocha#timeout

**Comment by vaucouleur** [comment]
@mleanos  Obviously, we can have both. Instead of mochajs default (2000), we can have a default value that fits better MEANJS needs, and if needed we can adjust the timeout for a specific test.

**Comment by mleanos** [comment]
@vaucouleur I agree. I've tested this for Grunt & Gulp & they seem to work as expected. I'm fine with having this option added.

**Comment by lirantal** [comment]
I think @vaucouleur has a valid point, let's add it as a global parameter for now instead of fiddling with each specific test scenario.

**Comment by mleanos** [comment]
@vaucouleur I can submit a PR for this, unless you're planning to. 

Should we remove the individual test suite timeouts when we add this?

**Comment by lirantal** [comment]
@mleanos @vaucouleur I think it's worth another PR change to move this to the grunt configuration.

**Comment by vaucouleur** [comment]
@mleanos Hi Michael, please go ahead with the PR. If it turns out that you're busy with other things, just let me know and I'll do it.

**Comment by mleanos** [comment]
This can be closed. I did a few rounds of testing after #972 was merged, and both Grunt & Gulp are working with this change.
